packages/cxl-ui/src/components/cxl-app-layout.js

```diff
@@ -11,22 +11,23 @@ import '@vaadin/vaadin-context-menu/src/vaadin-device-detector.js';
 import '@vaadin/vaadin-tabs';
 import '@vaadin/vaadin-progress-bar';
 
-const ASIDE_LOCAL_STORAGE_KEY = 'cxl-app-layout-aside-opened';
-
 @customElement('cxl-app-layout')
 export class CXLAppLayout extends LitElement {
   @query('aside')
   asideElement;
 
+  @property({ type: String, attribute: 'aside-local-storage-key' })
+  _asideLocalStorageKey = `cxl-app-layout-aside-opened--${window.location.pathname}`;
```

Comment:
This change ensures that different pages have their aside opened status registered separately. It had to become a class observed property to work properly.

Reply:
I'm not convinced here.

Reply:
> This change ensures that different pages have their aside opened status registered separately.

Why is this necessary? Wouldn't it become management hell for the user, sidebar jumping back and forth when browsing around pages?

Reply:
While testing, I found the opposite occurred. Let's say we have two different pages: a course page with the lessons list, and an unrelated page, like a blog, which would use the reversed layout (2c-l), on different tabs. Maybe the user likes to have the lessons opened all the time, but on the other page, the aside element has a related articles list, which the user doesn't want open. Going back and forth between them, the user would ever be allowed to have the two sidebars stay in opposite states. That's why I made it possible for them to be independent.

Reply:
Additionally, because it is now a litElement class property, it can be set with any value declaratively, via attribute. So, we can set a common key for all courses, for example, which would maintain a consistent behavior for pages with similar contexts. 

That's my rationale, not an ideal design description. If you tell me the shared global state is better for the way the CXL site is used, I'll just roll it back.

Reply:
@lkraav Should I revert this?

Reply:
I'd love to explore this in a separate PR.

There are most definitely cases where we should declaratively force it one way or another. I just don't have the specific use case in my head right now, and in the interest in keeping every PR surface as small as we can, we should exclude it here.